### ISSUE TITLE
v19 profile seperation for addon_data is not working correctly?

### ISSUE THREAD

**Issue by smarthomejarvis** [xbmc/xbmc#18760]
if you create 2 new profiles in kodi.
each profile should have seperated sources/databases

profiles are created, with the follwing structure, seems OK
```
~/.kodi/userdata/profiles$ tree -L 2
.
├── empty
│   ├── addon_data
│   ├── Database
│   ├── favourites.xml
│   ├── guisettings.xml
│   ├── keymaps
│   ├── library
│   ├── peripheral_data
│   ├── playlists
│   ├── Savestates
│   ├── sources.xml
│   └── Thumbnails
├── test
│   ├── addon_data
│   ├── Database
│   ├── favourites.xml
│   ├── guisettings.xml
│   ├── keymaps
│   ├── library
│   ├── peripheral_data
│   ├── playlists
│   ├── Savestates
│   ├── sources.xml
│   └── Thumbnails
└── tv
    ├── addon_data
    ├── Database
    ├── favourites.xml
    ├── guisettings.xml
    ├── keymaps
    ├── library
    ├── peripheral_data
    ├── playlists
    ├── Savestates
    ├── sources.xml
    └── Thumbnails
```

configuration for pvr.iptvsimple seems, OK
```
ls {tv,test,empty}/addon_data/pvr.iptvsimple/settings.xml
empty/addon_data/pvr.iptvsimple/settings.xml  test/addon_data/pvr.iptvsimple/settings.xml  tv/addon_data/pvr.iptvsimple/settings.xml
```
BUT, it seems like that the profile, which is configured as autologin, is used for all other profiles. The settings.xml for addon simpleiptv in .../pvr.iptvsimple/settings.xml is not used

nearby my settings.xml
```
<settings version="2">
    <setting id="m3uPathType">0</setting>
    <setting id="m3uPath">THIS_IS_A_LOCAL_PATH_REMOVED_FOR_GIT.m3u8</setting>
    <setting id="m3uUrl" default="true" />
    <setting id="m3uCache" default="true">false</setting>
    <setting id="startNum" default="true">1</setting>
    <setting id="numberByOrder" default="true">false</setting>
    <setting id="m3uRefreshMode">2</setting>
    <setting id="m3uRefreshIntervalMins" default="true">60</setting>
    <setting id="m3uRefreshHour" default="true">4</setting>
    <setting id="epgPathType">0</setting>
    <setting id="epgPath">THIS_IS_A_LOCAL_PATH_REMOVED_FOR_GIT</setting>
    <setting id="epgUrl" default="true" />
    <setting id="epgCache" default="true">true</setting>
    <setting id="epgTimeShift" default="true">0</setting>
    <setting id="epgTSOverride" default="true">false</setting>
    <setting id="useEpgGenreText">true</setting>
    <setting id="genresPathType" default="true">0</setting>
    <setting id="genresPath" default="true">special://userdata/addon_data/pvr.iptvsimple/genres/genreTextMappings/genres.xml</setting>
    <setting id="genresUrl" default="true" />
    <setting id="logoPathType" default="true">1</setting>
    <setting id="logoPath" default="true" />
    <setting id="logoBaseUrl" default="true" />
    <setting id="logoFromEpg" default="true">1</setting>
    <setting id="timeshiftEnabled" default="true">false</setting>
    <setting id="timeshiftEnabledAll" default="true">true</setting>
    <setting id="timeshiftEnabledHttp" default="true">true</setting>
    <setting id="timeshiftEnabledUdp" default="true">true</setting>
    <setting id="timeshiftEnabledCustom" default="true">false</setting>
    <setting id="catchupEnabled" default="true">false</setting>
    <setting id="catchupQueryFormat" default="true" />
    <setting id="catchupDays" default="true">5</setting>
    <setting id="allChannelsCatchupMode" default="true">0</setting>
    <setting id="catchupPlayEpgAsLive" default="true">false</setting>
    <setting id="catchupWatchEpgBeginBufferMins" default="true">5</setting>
    <setting id="catchupWatchEpgEndBufferMins" default="true">15</setting>
    <setting id="catchupOnlyOnFinishedProgrammes" default="true">false</setting>
    <setting id="transformMulticastStreamUrls" default="true">false</setting>
    <setting id="udpxyHost" default="true">127.0.0.1</setting>
    <setting id="udpxyPort" default="true">4022</setting>
    <setting id="useFFmpegReconnect" default="true">true</setting>
    <setting id="useInputstreamAdaptiveforHls" default="true">false</setting>
    <setting id="defaultUserAgent" default="true" />
    <setting id="defaultInputstream" default="true" />
    <setting id="defaultMimeType" default="true" />
</settings>
```

**Comment by DaVukovic** [comment]
Please use the forums

We don't use a profile which is called "empty". This might be the "master" profile and you might have renamed it. Anyway, there's a high recommendation to not use or install any add-ons using the "master"-profile if your plan is to use other profiles with the same add-on and different settings. 

Furthermore it all depends on how you switch from one profile to the other. The add-on data and its settigns are read. But if you load profile "A" and then switch to profile "B" via "settings -> profiles -> <profile-name>" then only 'new' settings are being loaded and previously loaded settings are overwritten if the new settings do have a different value for that setting. Otherwise, if there's no new value for that setting, the previously loaded value is still valid for the other profile. 

I would also not recommend to use "auto login" if you have the same add-ons for different profiles. Use the login-screen instead and choose the profile to be used. The "problem" is, that the master profile is always loaded no matter if you use that or not. So settings from that profile are loaded as well. 

To sum it up: 
- don't use the master profile at all
-  as this add-on here is a binary add-on and might be installed on the master profile, simply disable the add-on at the master profile
- don't use auto-log, use the login-screen instead.

We can't say more, because there's a bunch of information missing and you missed to provide a debuglog. But I highly guess stuff like this can be handled at the forums as it's mostly a user-error ;)

**Comment by ksooo** [comment]
Try latest v19 nightly build. This bug has been fixed recently.
